ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -7175,6 +7176,23 @@ private ExprNodeDesc getNotNullConstraintExpr(Table targetTable, Operator input,
     return currUDF;
   }
 
+  private Path getDestinationFilePath(final String destinationFile, boolean isMmTable)
+      throws SemanticException {
+    if (this.isResultsCacheEnabled() && this.queryTypeCanUseCache()) {
+      assert (!isMmTable);
```

Comment:
Is isMmTable necessarily false when this method is called below?

Reply:
From what I understood this will either be called for SELECT queries or CTAS. Result cached is disabled for CTAS. This is kind of indirect way of asserting that if result cache is enabled it is SELECT and not CTAS.